spec/features/signing_in_spec.rb

```diff
@@ -0,0 +1,21 @@
+feature 'A user can sign up' do
+  scenario 'user sees first page' do
+    visit '/'
+    expect(page).to have_content 'Welcome'
+  end
+
+  scenario 'user can sign up' do
+    sign_up
+  end
+
+  scenario 'user can sign in' do
+    sign_in
+    expect(page).to have_content 'Welcome, foobar_user'
+  end
+
+  scenario 'I cannot sign up with an existing email' do
+    sign_up
```

Comment:
You only need to create the user in the database here - not go through the whole sign up process